packages/web/src/pages/audio-rewards-page/components/modals/ChallengeRewardsModal/ChallengeRewardsModal.tsx

```diff
@@ -380,113 +383,133 @@ const ChallengeRewardsBody = ({ dismissModal }: BodyProps) => {
-      errorContent={errorContent}
-      onNavigateAway={dismissModal}
-      onClaimRewardClicked={onClaimRewardClicked}
-      claimInProgress={claimInProgress}
-      challenge={challenge}
-      challengeName={modalType}
-    />
-  ) : (
-    <div className={wm(styles.container)}>
-      {isMobile ? (
-        <>
-          {progressDescription}
-          <div className={wm(styles.progressCard)}>
-            <div className={wm(styles.progressInfo)}>
+  const { isEnabled: isRewardsCooldownEnabled } = useFeatureFlag(
+    FeatureFlags.REWARDS_COOLDOWN
+  )
+  if (isRewardsCooldownEnabled && challenge && challenge.cooldown_days > 0) {
+    return (
+      <CooldownRewardsModalContent
+        errorContent={errorContent}
+        onNavigateAway={dismissModal}
+        onClaimRewardClicked={onClaimRewardClicked}
+        claimInProgress={claimInProgress}
+        challenge={challenge}
+        challengeName={modalType}
+        onClickProgress={goToRoute}
+        progressIcon={buttonInfo?.rightIcon}
+        progressLabel={buttonInfo?.label}
+      />
+    )
+  } else if (!isRewardsCooldownEnabled && isAudioMatchingChallenge(modalType)) {
+    return (
+      <AudioMatchingRewardsModalContent
+        errorContent={errorContent}
+        onNavigateAway={dismissModal}
+        onClaimRewardClicked={onClaimRewardClicked}
+        claimInProgress={claimInProgress}
+        challenge={challenge}
+        challengeName={modalType}
+      />
+    )
+  } else {
+    return (
+      <div className={wm(styles.container)}>
+        {isMobile ? (
+          <>
+            {progressDescription}
+            <div className={wm(styles.progressCard)}>
+              <div className={wm(styles.progressInfo)}>
+                {progressReward}
+                {showProgressBar ? (
+                  <div className={wm(styles.progressBarSection)}>
+                    <h3>Progress</h3>
+                    <ProgressBar
+                      className={wm(styles.progressBar)}
+                      value={currentStepCount}
+                      max={challenge?.max_steps}
+                    />
+                  </div>
+                ) : null}
+              </div>
+              {progressStatusLabel}
+            </div>
+            {modalType === 'profile-completion' && <ProfileChecks />}
+          </>
+        ) : (
+          <div className={styles.progressCard}>
+            <div className={styles.progressInfo}>
+              {progressDescription}
               {progressReward}
-              {showProgressBar ? (
-                <div className={wm(styles.progressBarSection)}>
-                  <h3>Progress</h3>
-                  <ProgressBar
-                    className={wm(styles.progressBar)}
-                    value={currentStepCount}
-                    max={challenge?.max_steps}
-                  />
-                </div>
-              ) : null}
             </div>
+            {showProgressBar && (
```

Comment:
Heads up we prefer ternary like it was before

Reply:
oh i actually didn't change this but i can make it a ternary